src/types/uint.rs

```diff
@@ -93,6 +115,16 @@ macro_rules! impl_uint {
       }
     }
 
+    impl<'a> From<&'a [u8]> for $name {
+      fn from(num: &'a [u8]) -> Self {
+        let mut arr = [0; $len];
+        for i in 0..cmp::min(num.len(), $len) {
+          arr[$len - 1 - i] = num[num.len() - 1 - i]
```

Comment:
seems like this whole thing can be replaced with a `arr[..].copy_from_slice(&x[..len])` which will do a direct `memcpy` and not rely on LLVM optimizing the loop